### PR TITLE
Adding OFFICER as a pr annouce medium

### PR DESCRIPTION
**Pull request by trumpetx**
Recently we've decided to move toward a more "hidden bid" system and the only option was WHISPER which worked well enough, but we wanted the officers to see the bids too:

image

### CODE REVIEW

**Comment by trumpetx** [comment]
I also added another feature:  Customizable "announce text"

image

**Comment by trumpetx** [comment]
... And fix the /epgp config not opening up the config screen on the first try

File: libs/LibEPGPChat-1.0.lua [hunk 1]
```diff
@@ -4,6 +4,10 @@ local MINOR_VERSION = 10000
 local lib, oldMinor = LibStub:NewLibrary(MAJOR_VERSION, MINOR_VERSION)
 if not lib then return end
 
+function lib:Interp(s, tab)
+  return (s:gsub('($%b{})', function(w) return tab[w:sub(3, -2)] or w end))
+end
```
Comment (jpxthu) [inline libs/LibEPGPChat-1.0.lua:9]: Please put this function between `Announce` and `Whisper` according to alphabetical order.

File: epgp_options.lua [hunk 2]
```diff
@@ -103,6 +104,11 @@ function EPGP:ProcessCommand(str)
   str = str:gsub("%%t", UnitName("target") or "notarget")
   local command, nextpos = self:GetArgs(str, 1)
   if command == "config" then
+    if self.firstOpen then
+      -- long standing WOW interface bug where the config does not open the first time you execute it
+      self.firstOpen = false
+      InterfaceOptionsFrame_OpenToCategory("EPGP")
+    end
```
Comment (jpxthu) [inline epgp_options.lua:111]: I remembered this bug. Thanks.

File: localization/localization.zhCN.lua [hunk 3]
```diff
@@ -88,6 +88,8 @@ L["Do you want to resume recurring award (%s) %d EP/%s?"] = "你想重新开始
 L["EP Reason"] = "EP 理由"
 L["EP/GP are reset"] = "EP/GP 已被重置"
 L["EP/GP/PR announce medium"] = "EP/GP/PR 通告媒介"
+L["EP/GP/PR announce text"] = "EP/GP/PR announce text"
+L["Default:\n${char} EP:${ep} GP:${gp} PR:${pr}\n\nAdditional Variables:\n${bid}"] = "Default:\n${char} EP:${ep} GP:${gp} PR:${pr}\n\nAdditional Variables:\n${bid}"
```
Comment (jpxthu) [inline localization/localization.zhCN.lua:92]: You don't have to modify this file. Just leave these two items blank.

File: libs/LibEPGPChat-1.0.lua (hunk 1 above)
Comment (trumpetx) [reply libs/LibEPGPChat-1.0.lua]: done!

File: localization/localization.zhCN.lua (hunk 3 above)
Comment (trumpetx) [reply localization/localization.zhCN.lua]: I was just following the pattern from the other text there - I wasn't sure if it was a "reminder" to translate later or whether to just let the i18n file take care of itself.  Sounds good to me!

**Comment by trumpetx** [comment]
> Thanks for your contribution. I will make a short test before publish a new release.

I plan on adding a couple other additions this weekend if you wanted to wait, but they're a little bigger and might be better added separately.

**Comment by jpxthu** [comment]
@trumpetx It is better to separate into small PRs.

By the way, another person is working on auto EP reward on boss kill: https://github.com/jpxthu/EPGP-Classic/pull/37

**Pull request merged**